internal/webhooks/cluster_test.go

```diff
@@ -1456,7 +1456,7 @@ func TestClusterTopologyValidationWithClient(t *testing.T) {
 			wantErr:         false,
```

Comment:
Shouldn't l.1385 be wantErr: false now?

At a first glance seems wrong that we get an error in this case

Reply:
Ohhh!

If I omit the unhealthy conditions, the error goes away. 

```
WithControlPlaneMachineHealthCheck(&clusterv1.MachineHealthCheckTopology{
	MachineHealthCheckClass: clusterv1.MachineHealthCheckClass{},
}).
```

Reply:
I don't know, is that empty condition should be allowed?

Reply:
find it: https://github.com/kubernetes-sigs/cluster-api/blob/235d965810ba2a29aec88ec5d34291e82e6f8891/internal/webhooks/cluster.go#L398-L466

Reply:
specifically:
 https://github.com/kubernetes-sigs/cluster-api/blob/235d965810ba2a29aec88ec5d34291e82e6f8891/internal/webhooks/cluster.go#L404-L415

Reply:
Okay. I think it is a valid case that nothing is configured. In this case we still get MHC remediations if the node goes away (we don't need any nodeStartupTimeout or unhealthyConditions for that).

The test is failing for a different reason as you showed.

Let's add the following in l.1383 and change wantErr to false and adjust the name in 1368 accordingly
```go
WithControlPlaneInfrastructureMachineTemplate(&unstructured.Unstructured{}).
```

Reply:
Also tested with an empty machineHealthCheck object (after defaulting in the MHC webhook) and I get the following MHC In a cluster (which seems useful):

```yaml
spec:
  clusterName: my-cluster
  maxUnhealthy: 100%
  nodeStartupTimeout: 10m0s
  selector:
    matchLabels:
      topology.cluster.x-k8s.io/deployment-name: default-worker-topo-2
      topology.cluster.x-k8s.io/owned: ""
```

Reply:
ready